lib/_http_common.js

```diff
@@ -195,3 +195,13 @@ function httpSocketSetup(socket) {
   socket.on('drain', ondrain);
 }
 exports.httpSocketSetup = httpSocketSetup;
+
+/**
+ * Verifies that the given val is a valid HTTP token
+ * per the rules defined in RFC 7230
+ **/
+const token = /^[a-zA-Z0-9\!\#\$\%\&\'\*\+\-\.\^\_\`\|\~]+$/;
```

Comment:
We don't have to use a lot of escaping. Just move `-` to the end and we are safe to write the same like this `/^[\w!#$%&'*+.^`|~-]+$/`